sklearn/tree/tests/test_tree.py

```diff
@@ -1609,3 +1613,27 @@ def test_mae():
     dt_mae.fit([[3],[5],[3],[8],[5]],[6,7,3,4,3], [0.6,0.3,0.1,1.0,0.3])
     assert_array_equal(dt_mae.tree_.impurity, [7.0/2.3, 3.0/0.7, 4.0/1.6])
     assert_array_equal(dt_mae.tree_.value.flat, [4.0, 6.0, 4.0])
+
+
+def test_criterion_copy():
+    # Let's check whether copy of our criterion has the same type
+    # and properties as original
+    n_outputs = 3
+    n_classes = np.arange(3)
+    n_samples = 100
+    for copy_func in [copy.copy, copy.deepcopy]:
+        for _, typename in CRITERIA_CLF.items():
+            criteria = typename(n_outputs, n_classes)
+            result = copy_func(criteria).__reduce__()
```

Comment:
Rather could you pickle, load all the Criteria?

Currently this fails in master...

```python
import pickle

from sklearn.tree._criterion import FriedmanMSE

fmse = FriedmanMSE(1, 10)
pickle.loads(pickle.dumps(fmse))
```